### PR TITLE
[core] Theming improvements

### PR DESCRIPTION
**Pull request by marechal-p**
- Make the theme change a bit smarter and only trigger the update if a new theme is being set.
- Fix a typo.

Relative to https://github.com/theia-ide/theia/issues/2503#issuecomment-411324372.

Signed-off-by: Paul Maréchal <paul.marechal@ericsson.com>

### CODE REVIEW

File: packages/core/src/browser/theming.ts [hunk 1]
```diff
@@ -108,10 +111,10 @@ export class ThemeService {
     }
 
     /**
-     * The default them. If that is not applicable, returns with the fallback theme.
+     * The default theme. If that is not applicable, returns with the fallback theme.
      */
     get defaultTheme(): Theme {
-        return this.themes[this._defaultTheme || this.fallbackTheme] || this.themes[this.fallbackTheme];
+        return this.themes[this._defaultTheme || this.fallbackTheme];
```
Comment (kittaakos) [inline packages/core/src/browser/theming.ts]: Why did you remove it?

Comment (marechal-p) [reply packages/core/src/browser/theming.ts]: Because while fixing the typo in the comment, it seemed to me to be useless ?

If `this._defaultTheme` is empty/undefined, *it will* use `this.fallbackTheme`.

So the expression will already be equivalent to `this.themes[this._defaultTheme]`, so writing it a second time felt weird to me.

Comment (kittaakos) [reply packages/core/src/browser/theming.ts]: If `this._defaultTheme` is defined `"foo"` but there is no such theme we have to fall back to the fallback. Please revert.

Comment (marechal-p) [reply packages/core/src/browser/theming.ts]: thanks

Comment (kittaakos) [reply packages/core/src/browser/theming.ts]: No. https://github.com/theia-ide/theia/issues/2391

Comment (marechal-p) [reply packages/core/src/browser/theming.ts]: Also, I feel like the `fallbackTheme` is some kind of `defaultDefaultTheme` to the `_defaultTheme` member of the ThemeService object.

Is `FrontendApplicationConfigProvider` meant to be used by users or by developers ?

Because I would only rely on that, and get rid of the `fallbackTheme` and make it as a default to the frontend app config provider. Throw an error if default theme is not found.

Comment (marechal-p) [reply packages/core/src/browser/theming.ts]: It doesn't change the fact that users shouldn't touch the `package.json`, and setting an unknown theme as the default application theme is a mistake. Still not worth an error?

Comment (kittaakos) [reply packages/core/src/browser/theming.ts]: No.

Comment (marechal-p) [reply packages/core/src/browser/theming.ts]: Thanks for the feedback, but how about a `console.warn('defaulttheme not found')`?

Comment (kittaakos) [reply packages/core/src/browser/theming.ts]: It's up to you.

**Comment by marechal-p** [comment]
I will leave this PR as it is, as it already fixes the issue it was supposed to address in the first place.

@kittaakos is the PR in a good shape for you?

**Comment by kittaakos** [comment]
I am verifying it...
